autosklearn/util/data.py

```diff
@@ -135,8 +135,8 @@ def predict_RAM_usage(X: np.ndarray, categorical: List[bool]) -> float:
 
 
 def subsample(
-    X: SUPPORTED_FEAT_TYPES,
-    y: Union[List, np.ndarray, pd.DataFrame, pd.Series],
+    X: Union[np.ndarray, spmatrix],
```

Comment:
Wait, isn't that too narrow now? We can have dataframes here that require subsampling as only the pipeline converts to a numpy array.

Reply:
The use of this function is only `np.ndarray` and `spmatrix`, we can't determine a size for pandas dataframes yet so we do not call subsample with it. However, the function can support dataframes.

Reply:
So #1278 should be more general to reflect this fact, too?